Managers/FrameMan.cpp

```diff
@@ -1,2651 +1,1391 @@
+						}
+					}
+					yTextPos += 12;
+				}
+				g_PerformanceMan.Draw(pPlayerGUIBitmap);
+
+				// Draw info text when in MOID or material layer draw mode
+				switch (g_SceneMan.GetLayerDrawMode()) {
+					case g_LayerTerrainMatter:
+						GetSmallFont()->DrawAligned(&pPlayerGUIBitmap, GetPlayerScreenWidth() / 2, GetPlayerScreenHeight() - 12, "Viewing terrain material layer\nHit Ctrl+M to cycle modes", GUIFont::Centre, GUIFont::Bottom);
+						break;
+					case g_LayerMOID:
+						GetSmallFont()->DrawAligned(&pPlayerGUIBitmap, GetPlayerScreenWidth() / 2, GetPlayerScreenHeight() - 12, "Viewing MovableObject ID layer\nHit Ctrl+M to cycle modes", GUIFont::Centre, GUIFont::Bottom);
+						break;
+					default:
+						break;
+				}
 
+				// If superfluous screen (as in a three-player match), make the fourth the Observer one
+			} else {
+				//yTextPos += 12;
+				GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, GetPlayerScreenWidth() / 2, yTextPos, "- Observer View -", GUIFont::Centre);
+			}
 
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          DrawBitmapPrimitive
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Schedule to draw a bitmap primitive.
-// Arguments:       Player screen to draw primitive on. Position of primitive in scene coordintaes, an entity to sraw sprite from, 
-//					rotation angle in radians, frame to draw
-// Return value:    None.
-
-	void FrameMan::DrawBitmapPrimitive(int player, Vector start, Entity * pEntity, float rotAngle, int frame)
-	{
-		MOSprite * pMOS = dynamic_cast<MOSprite *>(pEntity);
-		if (pMOS)
-		{
-			BITMAP * pBitmap = pMOS->GetSpriteFrame(frame);
-			if (pBitmap)
-				m_Primitives.push_back(new BitmapPrimitive(player, start, pBitmap, rotAngle));
+			// If we are dealing with split screens, then deal with the fact that we need to draw the player screens to different locations on the final buffer
+			// The position of the current draw screen on the final screen
+			Vector screenOffset;
+
+			if (screenCount > 1) {
+				switch (whichScreen) {
+					case 1:
+						// If both splits, or just VSplit, then in upper right quadrant
+						if ((m_VSplit && !m_HSplit) || (m_VSplit && m_HSplit)) {
+							screenOffset.SetIntXY(GetResX() / 2, 0);
+						} else {
+							// If only HSplit, then lower left quadrant
+							screenOffset.SetIntXY(0, GetResY() / 2);
+						}
+						break;
+					case 2:
+						// Always lower left quadrant
+						screenOffset.SetIntXY(0, GetResY() / 2);
+						break;
+					case 3:
+						// Always lower right quadrant
+						screenOffset.SetIntXY(GetResX() / 2, GetResY() / 2);
+						break;
+					default:
+						// Always upper left corner
+						screenOffset.SetIntXY(0, 0);
+						break;
+				}
+			}
+			// Flash the screen if we're supposed to
```

Comment:
Several points about screen flashing:
1. All the comments in here are junk, except the one about `left`
2. `left` is a bad name. `timeTilLimit` or something more informative would bet better, without context left could be a direction or whatever.
3. Kill off the commented slowdown code. We definitely don't want that!
4. Pull this screen flashing stuff out into a separate private method.